src/DurableTask.AzureStorage/OrchestrationSessionManager.cs

```diff
@@ -174,6 +174,36 @@ async Task DequeueLoop(string partitionId, ControlQueue controlQueue, Cancellati
                 $"Stopped listening for messages on queue {controlQueue.Name}.");
         }
 
+        public async Task DrainAsync(string partitionId, CloseReason reason, CancellationToken cancellationToken)
+        {
+            // Start the drain process, mark the queue released to stop listening to new msg
+            this.ReleaseQueue(partitionId, reason, this.settings.WorkerId);
+            try
+            {
+                // Wait for drain to complete
```

Comment:
Assuming I understand this correctly, I think the following is more precise:
```suggestion
                // Wait until we've processed all messages from this queue
```